addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/handler/HomematicBridgeHandler.java

```diff
@@ -0,0 +1,307 @@
+     * Schedules a reinitialization, if the Homematic gateway is not reachable at bridge startup.
+     */
+    private void scheduleReinitialize() {
+        scheduler.schedule(new Runnable() {
+
+            @Override
+            public void run() {
+                initialize();
+            }
+        }, REINITIALIZE_DELAY_SECONDS, TimeUnit.SECONDS);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void dispose() {
+        logger.debug("Disposing bridge '{}'", getThing().getUID().getId());
+        super.dispose();
+        if (discoveryService != null) {
+            discoveryService.stopLoadDevices();
+            unregisterDeviceDiscoveryService();
+        }
+        if (gateway != null) {
+            gateway.dispose();
+        }
+        if (config != null) {
+            portPool.release(config.getCallbackPort());
+        }
+    }
+
+    /**
+     * Registers the DeviceDisoveryService.
+     */
+    private void registerDeviceDiscoveryService() {
+        logger.trace("Registering HomematicDeviceDiscoveryService for bridge '{}'", getThing().getUID().getId());
+        discoveryService = new HomematicDeviceDiscoveryService(this);
+        discoveryServiceRegistration = bundleContext.registerService(DiscoveryService.class.getName(), discoveryService,
+                new Hashtable<String, Object>());
+        discoveryService.activate();
+    }
+
+    /**
+     * Unregisters the DeviceDisoveryService.
+     */
+    private void unregisterDeviceDiscoveryService() {
+        if (discoveryServiceRegistration != null) {
+            HomematicDeviceDiscoveryService service = (HomematicDeviceDiscoveryService) bundleContext
+                    .getService(discoveryServiceRegistration.getReference());
+            service.deactivate();
+
+            discoveryServiceRegistration.unregister();
+            discoveryServiceRegistration = null;
+            discoveryService = null;
+        }
+    }
+
+    /**
+     * Sets the OFFLINE status for all things of this bridge that has been removed from the gateway.
+     */
+    public void setOfflineStatus() {
+        for (Thing hmThing : getThing().getThings()) {
+            try {
+                gateway.getDevice(UidUtils.getHomematicAddress(hmThing));
+            } catch (HomematicClientException e) {
+                ((HomematicThingHandler) hmThing.getHandler()).updateStatus(ThingStatus.OFFLINE);
+            }
+        }
+    }
+
+    /**
+     * Creates the configuration for the HomematicGateway.
+     */
+    private HomematicConfig createHomematicConfig() {
+        HomematicConfig homematicConfig = getThing().getConfiguration().as(HomematicConfig.class);
+        if (homematicConfig.getCallbackHost() == null) {
+            homematicConfig.setCallbackHost(LocalNetworkInterface.getLocalNetworkInterface());
+        }
+        if (homematicConfig.getCallbackPort() == 0) {
+            homematicConfig.setCallbackPort(portPool.getNextPort());
+        }
```

Comment:
I believe you also need an else branch here: If a port is specified, you need to declare it used within the portPool as otherwise the next handler without a port might be assigned the same port number.